### PR TITLE
First drop of chunk_get proof-of-concept.

### PR DESCRIPTION
**Pull request by richkong88**
Just a proof-of-concept on chunk_get implementation to get ideas. Please do NOT commit.

Signed-off-by: Kong, Richard <richard.kong@intel.com>

### CODE REVIEW

File: include/library/spdm_lib_config.h [hunk 1]
```diff
@@ -270,12 +275,17 @@
...
+ * If chunking is supported, it should be at least LIBSPDM_SENDER_RECEIVE_BUFFER_SIZE
+ * LIBSPDM_MAX_SPDM_MSG_SIZE.
  *
  * The value is NOT configurable.
- * The value MAY be chnaged in different libspdm version.
+ * The value MAY be changed in different libspdm version.
  * It is exposed here, just in case the libspdm consumer wants to configure the setting at build time.
  */
-#define LIBSPDM_SCRATCH_BUFFER_SIZE (LIBSPDM_SENDER_RECEIVE_BUFFER_SIZE)
+#define LIBSPDM_SCRATCH_BUFFER_SIZE (LIBSPDM_SENDER_RECEIVE_BUFFER_SIZE + LIBSPDM_MAX_SPDM_MSG_SIZE)
```
Comment (jyao1) [inline include/library/spdm_lib_config.h:289]: do we need include LIBSPDM_MAX_SPDM_MSG_SIZE if LIBSPDM_ENABLE_CHUNK_CAP is 0 ?

**Comment by jyao1** [comment]
Overall, it looks good.

Question: How do we trigger the flow in unit test?

File: include/library/spdm_lib_config.h (hunk 1 above)
Comment (richkong88) [reply include/library/spdm_lib_config.h:289]: You are right. LIBSPDM_SCRATCH_BUFFER should depend on if LIBSPDM_ENABLE_CHUNK_CAP is enabled or not.

**Comment by richkong88** [comment]
> Overall, it looks good.
> 
> Question: How do we trigger the flow in unit test?

Thanks. Good question. I'm still trying to figure this part out. I was just testing by modifying the spdm_lib_config.h parameters (eg. LIBSPDM_DATA_TRANSFER_SIZE) so it would force larger responses to go through the CHUNK_GET flow.  Is it possible to dynamically override some of these defines when compiling for ULT?

**Comment by jyao1** [comment]
> Is it possible to dynamically override some of these defines when compiling for ULT?

Sure, we can override.
In my definition, dynamic == runtime override, static == compiling time override.
Would you please clarify what you mean when you have both "dynamically" and "compiling" ?

**Comment by richkong88** [comment]
> > Is it possible to dynamically override some of these defines when compiling for ULT?
> 
> Sure, we can override. In my definition, dynamic == runtime override, static == compiling time override. Would you please clarify what you mean when you have both "dynamically" and "compiling" ?

I meant force the compiler to change the values for the LIBSPDM_DATA_TRANSFER_SIZE define without modifying the code. I think we can do this with the -DLIBSPDM_DATA_TRANSFER_SIZE=X, since the define is protected with #ifndef #endif in the spdm_lib_config file. I think we'd want to run the ULTs with their default values, then run them again with the config settings to force exercising the chunking flows.

**Comment by jyao1** [comment]
> I think we can do this with the -DLIBSPDM_DATA_TRANSFER_SIZE=X

yes we can

**Comment by richkong88** [comment]
Looking more at unit tests now. This is more complicated than just changing a define. We'll have to write a bunch of unit_test cases. I'll work on this.

**Comment by richkong88** [comment]
Closing since this was just POC to get feedback.

**Pull request closed without merging**